controllers/profiles/following.js

```diff
@@ -35,16 +49,16 @@ module.exports = async (req, res) => {
       "user"."karma_score" AS "user.karma_score",
       ARRAY( select name from topics as tp
         left join user_topics as utp on tp.topic_id = utp.topic_id
-        where utp.user_id = "user".user_id limit 3
+        where utp.user_id = "user".user_id and tp.topic_id in (:topicIds) limit 3
       ) as "user.community_info"
       FROM "user_follow_user" AS "UserFollowUser"
       LEFT OUTER JOIN 
       "users" AS "user" ON "UserFollowUser"."user_id_followed" = "user"."user_id"
-      WHERE "UserFollowUser"."user_id_follower" = :id AND "UserFollowUser"."user_id_followed" != :id AND "user"."is_anonymous" = false;`,
+      WHERE "UserFollowUser"."user_id_follower" = :userId AND "UserFollowUser"."user_id_followed" != :userId AND "user"."is_anonymous" = false;`,
     {
       nest: true,
       type: QueryTypes.SELECT,
-      replacements: {id}
+      replacements: {userId, topicIds}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This part of the code seems fine. It correctly uses the `topicIds` obtained from the previous query to filter the community info for the user. However, ensure that the `topicIds` are properly sanitized or parameterized to prevent SQL injection attacks.
```diff
+ where utp.user_id = "user".user_id and tp.topic_id in (:topicIds) limit 3
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->